source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1896 @@
+`DOCS-10784`_); however, there is scattered documentation available on the
+server wiki (`The "failCommand" Fail Point <failpoint-wiki_>`_) and employee blogs
+(e.g. `Intro to Fail Points <failpoint-blog1_>`_,
+`Testing Network Errors with MongoDB <failpoint-blog2_>`_). Documentation can
+also be gleaned from JIRA tickets (e.g. `SERVER-35004`_, `SERVER-35083`_). This
+specification does not aim to provide comprehensive documentation for all fail
+points available for driver testing, but some fail points are documented in
+`Fail Points Commonly Used in Tests`_.
+
+.. _failpoint-wiki: https://github.com/mongodb/mongo/wiki/The-%22failCommand%22-fail-point
+.. _failpoint-blog1: https://kchodorow.com/2013/01/15/intro-to-fail-points/
+.. _failpoint-blog2: https://emptysqua.re/blog/mongodb-testing-network-errors/
+.. _DOCS-10784: https://jira.mongodb.org/browse/DOCS-10784
+.. _SERVER-35004: https://jira.mongodb.org/browse/SERVER-35004
+.. _SERVER-35083: https://jira.mongodb.org/browse/SERVER-35083
+
+
+Configuring Fail Points
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``configureFailPoint`` command should be executed on the ``admin`` database
+and has the following structure::
+
+    db.adminCommand({
+        configureFailPoint: <string>,
+        mode: <string|document>,
+        data: <document>
+    });
+
+The value of ``configureFailPoint`` is a string denoting the fail point to be
+configured (e.g. "failCommand").
+
+The ``mode`` option is a generic fail point option and may be assigned a string
+or document value. The string values "alwaysOn" and "off" may be used to
+enable or disable the fail point, respectively. A document may be used to
+specify either ``times`` or ``skip``, which are mutually exclusive:
+
+- ``{ times: <integer> }`` may be used to limit the number of times the fail
+  point may trigger before transitioning to "off".
+- ``{ skip: <integer> }`` may be used to defer the first trigger of a fail
+  point, after which it will transition to "alwaysOn".
+
+The ``data`` option is a document that may be used to specify any options that
+control the particular fail point's behavior.
+
+In order to use ``configureFailPoint``, the undocumented ``enableTestCommands``
+`server parameter <https://docs.mongodb.com/manual/reference/parameters/>`_ must
+be enabled by either the configuration file or command line option (e.g.
+``--setParameter enableTestCommands=1``). It cannot be enabled at runtime via
+the `setParameter <https://docs.mongodb.com/manual/reference/command/setParameter/>`_
+command). This parameter should already be enabled for most configuration files
+in `mongo-orchestration <https://github.com/10gen/mongo-orchestration>`_.
+
+
+Disabling Fail Points
+~~~~~~~~~~~~~~~~~~~~~
+
+A fail point may be disabled like so::
+
+    db.adminCommand({
+        configureFailPoint: <string>,
+        mode: "off"
+    });
+
+
+Fail Points Commonly Used in Tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+failCommand
```

Comment:
Does it make sense to replicate this here rather than just rely on the docs for it in the server wiki? Seems like a maintenance headache though it certainly makes it easier to implement the spec by having the info in one place

Reply:
I think when it comes to fail point documentation, we're better off documenting as much of that as possible in a tracked repository. The server wiki is nice, but we don't own that and it reminds me of when `configureFailPoint` was pulled out of the server documentation (which is why those pages only exist in some foreign language copies of our old docs now).